### PR TITLE
Create status labels

### PR DESCRIPTION
**Pull request by BenjaminParment**
-> Adds status labels for each different status state.
-> Adds tests for the factory
-> Border css has been tested in all browsers

image

### CODE REVIEW

File: src/components/reusables/status-label/StatusLabel.scss [hunk 1]
```diff
@@ -0,0 +1,45 @@
+@import "~shared/style/mixins";
+
+$STATUS_ORANGE_COLOR: #FEF4E2;
+$STATUS_GREEN_COLOR: #E9F9EA;
+$STATUS_RED_COLOR: #FFE5E4;
+$STATUS_GREY_COLOR:#EBEBEB;
```
Comment (j-n-white) [inline src/components/reusables/status-label/StatusLabel.scss]: Keep the naming design agnostic. If we decide to change the cancelled colour to blue for example we should not have to rename any variables.

File: src/components/reusables/status-label/StatusLabelFactory.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        case "Cancelled":
+            return (
+                <StatusLabel
+                    statusText={type}
+                    styleClass={"status-cancel"}
+                    icon={<BlockIcon height={iconDimension} width={iconDimension} />}
+                />
+            );
+        default:
+            return <div>Wrong Status Label</div>;
```
Comment (j-n-white) [inline src/components/reusables/status-label/StatusLabelFactory.tsx]: Don't return a random div throw an error.

File: src/components/reusables/status-label/StatusLabel.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import "./StatusLabel.scss";
+
+import React from "react";
+
+export type LabelContent = "Pending approval" | "In Progress" | "Paid" | "Returned" | "Error" | "Cancelled";
+
+interface StatusLabelProps {
+    statusText: LabelContent;
+    styleClass: string;
+    icon: JSX.Element;
```
Comment (j-n-white) [inline src/components/reusables/status-label/StatusLabel.tsx]:
```suggestion
    icon: React.ComponentType<SVGProps>;

```
I think this should work.

File: src/components/reusables/status-label/StatusLabel.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import "./StatusLabel.scss";
+
+import React from "react";
+
+export type LabelContent = "Pending approval" | "In Progress" | "Paid" | "Returned" | "Error" | "Cancelled";
+
+interface StatusLabelProps {
+    statusText: LabelContent;
+    styleClass: string;
```
Comment (j-n-white) [inline src/components/reusables/status-label/StatusLabel.tsx]: Limit this type to just be one of the expected styles.

File: src/components/reusables/status-label/StatusLabelFactory.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import { ErrorOutlineIcon } from "resources/svgs/ErrorOutlineIcon";
+import { KeyboardReturnIcon } from "resources/svgs/KeyboardReturnIcon";
+import React from "react";
+import { SeeMoreHorizontalIcon } from "resources/svgs/SeeMoreHorizontalIcon";
+
+const iconDimension = 24;
+
+export const createStatusLabel = (type: LabelContent): JSX.Element => {
+    switch (type) {
+        case "Paid":
```
Comment (j-n-white) [inline src/components/reusables/status-label/StatusLabelFactory.tsx]: you could save a lot of duplication here if you just set the style and icon in the switch and then return a single status label.

File: src/components/reusables/status-label/StatusLabel.tsx (hunk 3 above)
Comment (BenjaminParment) [reply src/components/reusables/status-label/StatusLabel.tsx]: hmmm, it doesnt seem to work on my branch.. will look online

**Pull request merged**